nornir/core/inventory.py

```diff
@@ -449,6 +449,7 @@ def add_host(self, name: str, **kwargs) -> None:
             )
         }
         self.hosts.update(host)
+        self.__init__(hosts=self.hosts, groups=self.groups, defaults=self.defaults)
```

Comment:
I am concerned about side effects if the __init__ function changes or if the inventory is too large and this becomes  a waste of CPU cycles. Maybe we can just create a `_add_group_refs` function and explicitly execute these two lines:

```
def _update_group_refs(self, inventory_element: InventoryElement) -> None:
        inventory_element.groups.refs = [self.groups[p] for g in inventory_element.groups]
```

Or something similar.

Reply:
I added the helper and associated tests. It seems to work well like this.